### ISSUE TITLE
Corruption in test_small_copy

### ISSUE THREAD

**Issue by k4nar** [onitu/onitu#9]
I just got this fail in `test_small_copy` : 

```
============================================================================= FAILURES ==============================================================================
__________________________________________________________________________ test_small_copy __________________________________________________________________________

    def test_small_copy():
>       copy_file('simple', 10)

tests/functional/test_copy.py:55: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

filename = 'simple', size = 10

    def copy_file(filename, size):
        loop = BooleanLoop()
        launcher.on_transfer_ended(
            loop.stop, d_from='rep1', d_to='rep2', filename=filename
        )
        generate(os.path.join(rep1, filename), size)
        loop.run(timeout=5)
>       assert(checksum(os.path.join(rep1, filename)) ==
               checksum(os.path.join(rep2, filename)))
E       assert 'f73065c569ae...8f9607f736a72' == 'f7ce168fb71e8...f46423c5b87ae'
E         - f73065c569ae28fda418f9607f736a72
E         + f7ce168fb71e8609260f46423c5b87ae

tests/functional/test_copy.py:50: AssertionError
-------------------------------------------------------------------------- Captured stderr --------------------------------------------------------------------------
[16:55:02] INFO: Onitu: Loading setup...
[16:55:03] INFO: Referee: Started
[16:55:03] INFO: rep2: Started
[16:55:03] INFO: rep2 - Router: Started
[16:55:03] INFO: rep1: Started
[16:55:03] INFO: rep1 - Router: Started
[16:55:03] INFO: rep2 - Dealer: Started
[16:55:03] INFO: rep1 - Dealer: Started
[16:55:03] INFO: Referee: New event for 'simple' from rep1
[16:55:03] INFO: Referee: New event for 'simple' from rep1
[16:55:03] INFO: rep2 - Dealer: Starting to get 'simple' from rep1
[16:55:03] INFO: rep2 - Dealer: Starting to get 'simple' from rep1
[16:55:03] INFO: rep2 - Dealer: Transfer of 'simple' from rep1 successful
[16:55:03] INFO: rep2 - Dealer: Transfer of 'simple' from rep1 successful
=============================================================== 1 failed, 15 passed in 15.00 seconds ================================================================
```

The event for 'simple' is triggered twice. No idea why, and I never saw this before.

**Comment by k4nar** [comment]
Again, with debug : 

```
[22:37:33] INFO: Onitu: Loading setup...
[22:37:34] DEBUG: Onitu: Loading entry rep2
[22:37:34] DEBUG: Onitu: Loading entry rep1
[22:37:34] DEBUG: Onitu: Entries loaded
[22:37:34] INFO: Referee: Started
[22:37:34] INFO: rep2: Started
[22:37:34] INFO: rep2 - Router: Started
[22:37:34] INFO: rep1: Started
[22:37:34] INFO: rep1 - Router: Started
[22:37:34] INFO: rep2 - Dealer: Started
[22:37:34] INFO: rep1 - Dealer: Started
[22:37:34] DEBUG: rep1: Notifying the Referee about 'simple'
[22:37:34] DEBUG: rep1: Notifying the Referee about 'simple'
[22:37:34] INFO: Referee: New event for 'simple' from rep1
[22:37:34] DEBUG: Referee: Notifying rep2 about 'simple'
[22:37:34] INFO: Referee: New event for 'simple' from rep1
[22:37:34] DEBUG: Referee: Notifying rep2 about 'simple'
[22:37:34] INFO: rep2 - Dealer: Starting to get 'simple' from rep1
[22:37:34] DEBUG: rep2 - Dealer: Waiting for ROUTER port for rep1
[22:37:34] DEBUG: rep2 - Dealer: Got ROUTER port for rep1
[22:37:34] DEBUG: rep2 - Dealer: Connected
[22:37:34] DEBUG: rep2 - Dealer: Asking rep1 for a new chunk
[22:37:34] INFO: rep2 - Dealer: Starting to get 'simple' from rep1
[22:37:34] DEBUG: rep1 - Router: Getting chunk of size 1048576 from offset 0 in 'simple'
[22:37:34] DEBUG: rep2 - Dealer: Waiting for ROUTER port for rep1
[22:37:34] DEBUG: rep2 - Dealer: Received chunk of size 10 from rep1 for 'simple'
[22:37:34] DEBUG: rep2 - Dealer: Got ROUTER port for rep1
[22:37:34] DEBUG: rep2 - Dealer: Connected
[22:37:34] DEBUG: rep2 - Dealer: Asking rep1 for a new chunk
[22:37:34] DEBUG: rep1 - Router: Getting chunk of size 1048576 from offset 0 in 'simple'
[22:37:34] DEBUG: rep2 - Dealer: Received chunk of size 10 from rep1 for 'simple'
[22:37:34] INFO: rep2 - Dealer: Transfer of 'simple' from rep1 successful
[22:37:34] INFO: rep2 - Dealer: Transfer of 'simple' from rep1 successful
```

**Comment by k4nar** [comment]
I found the origin of the bug. This is happening if two threads are calling `Plug.update_file` for a new file at the same moment.
The bug happen more often since https://github.com/onitu/onitu/commit/50f1f7e6e5f6eebc6a67fa8b8c28cd39ae8a19b8. This is a side-effect of the fix of the issue #8.

I have a quick fix to return to the situation before https://github.com/onitu/onitu/commit/50f1f7e6e5f6eebc6a67fa8b8c28cd39ae8a19b8 (without breaking #8), but it will not properly fix the race-condition, only limit it. If after thousands of iterations I don't get any failure, I'll will close this issue.

**Comment by k4nar** [comment]
The quick fix wasn't enough, so I made a proper fix using Redis' transactions. It went ok ~3000 times.
